ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -415,48 +418,70 @@ private static void ensureKeySchemasMatch(
           + "operation potentially overrides existing key schema in schema registry.", e);
     }
 
-    if (latest.isPresent() && !latest.get().canonicalString().equals(schema.canonicalString())) {
-      final Map<String, String> formatProps = keyFormat.getFormatInfo().getProperties();
-
-      // Hack: skip comparing connect name. See https://github.com/confluentinc/ksql/issues/7211
-      // Avro schema are registered in source creation time as well data insertion time.
-      // CONNECT_META_DATA_CONFIG is configured to false in Avro Serializer, but it's true in
-      // AvroSchemaTranslator. It needs to be true to make ConnectSchema map type work. But
-      // enabling it breaks lots of history QTT test which implies backward compatibility issues.
-      // So we just bypass the connect name check here.
-      if (format instanceof AvroFormat) {
-        final SchemaTranslator translator = format.getSchemaTranslator(formatProps);
-        translator.configure(ImmutableMap.of(AvroDataConfig.CONNECT_META_DATA_CONFIG, false));
-        final ParsedSchema parsedSchema = translator.toParsedSchema(keySchema);
-        if (latest.get().canonicalString().equals(parsedSchema.canonicalString())) {
-          return;
-        }
-      } else if (format instanceof ProtobufFormat
-          && formatProps.containsKey(ConnectProperties.FULL_SCHEMA_NAME)) {
+    if (!latest.getSchema().isPresent()) {
+      throw new KsqlException("Failed to fetch schema from Schema Registry. Please try again.");
```

Comment:
Is there an error or a different action the user can take if this happens? Try again is because network issues? If the user sees this many times, they won't know what's wrong with SR.

Reply:
Btw, the old code does not throw an error if the schema is not present. Could this add some problems?

Reply:
The old code doesn't deal with this situation resulting in auto-registering a new schema. I thought we should prevent this from happening.

Reply:
Should we enhance the error message to say "Check Schema Registry connection"?

Reply:
Will create a separate issue for this as discussed.